src/tic-tac-toe.ts

```diff
@@ -40,36 +44,98 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinData {
+    const winData: WinData = {
+      hasWinner: false,
+      winningCells: [],
+    };
+
+    for (const rowIndex in this.board) {
```

Comment:
🚫 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in

Reply:
If you just want indices, you coooooould use `Object.entries` and destructure it off that. But agree 100% with @rylnd , and in particular, `in` gives you the stringy index instead, as does `Object.entries`.

```
for (const [indexAsString, _value] of Object.entries([1,2,3])) { 
  console.log(indexAsString); 
}
```